client/state/login/magic-login/selectors.js

```diff
@@ -3,10 +3,21 @@
  */
 import { get } from 'lodash';
 
-export const isShowingRequestForm = state => get( state, 'login.magicLogin.isShowingRequestForm', false );
-export const isShowingInterstitialPage = state => get( state, 'login.magicLogin.isShowingInterstitialPage', false );
-export const isShowingExpiredPage = state => get( state, 'login.magicLogin.isShowingExpiredPage', false );
-export const isShowingCheckYourEmailPage = state => get( state, 'login.magicLogin.isShowingCheckYourEmailPage', false );
+/**
+ * Internal dependencies
+ */
+import {
+	CHECK_YOUR_EMAIL_PAGE,
+	INTERSTITIAL_PAGE,
+	LINK_EXPIRED_PAGE,
+	REQUEST_FORM,
+} from './constants';
+
+export const showingView = state => get( state, 'login.magicLogin.showingView', null );
+export const isShowingCheckYourEmailPage = state => showingView( state ) === CHECK_YOUR_EMAIL_PAGE;
+export const isShowingInterstitialPage = state => showingView( state ) === INTERSTITIAL_PAGE;
+export const isShowingExpiredPage = state => showingView( state ) === LINK_EXPIRED_PAGE;
+export const isShowingRequestForm = state => showingView( state ) === REQUEST_FORM;
```

Comment:
Grepping reveals some of these selectors aren't actually in use (?). Given the recommendation to implement new selectors directly in `state/selectors`, one per file, I would consider:

- only implementing `showingView`, perhaps as `selectors/get-magic-login-current-view`
- having interested parties import `constants` and do their own eq check
- to alleviate the above, perhaps have the main view do something like

```js
const mapState = ( state ) => {
  const currentView = getMagicLoginCurrentView( state );
  return {
    isShowingYourEmailPage: currentView === CHECK_YOUR_EMAIL_PAGE,
    // and so on
  };
};
```

^ and hopefully that should be enough and those computed properties can be passed as props to other interested components.